satpy/writers/__init__.py

```diff
@@ -632,6 +632,11 @@ def create_filename_parser(self, base_dir):
             file_pattern = self.file_pattern
         return parser.Parser(file_pattern) if file_pattern else None
 
+    @staticmethod
+    def _prepare_metadata_for_filename_formatting(attrs):
+        if isinstance(attrs.get('sensor'), set):
```

Comment:
could `list` and `tuple` be actual too?

Reply:
In whatever kind of "standard" we have defined for Satpy's sensor field, no. We only ever have a string or a set. Could someone modify their data so it is a list or a tuple, yes, but I'm not sure we need to support that.

Reply:
ok